dspace-api/src/main/java/org/dspace/harvest/OAIHarvester.java

```diff
@@ -432,7 +432,7 @@ public void runHarvest() throws SQLException, IOException, AuthorizeException
 		log.info("Harvest from " + oaiSource + " successful. The process took " + timeTaken + " milliseconds. Harvested " + currentRecord + " items.");
 		harvestedCollection.update(ourContext, harvestRow);
 
-		ourContext.enableBatchMode(originalMode);
+		ourContext.setMode(originalMode);
```

Comment:
The difficulty with committing the transaction is that any reference to an entity from before the commit has to be reloaded from the database if we still intend to use them. This also applies to references in the calling method.